python/tank_vendor/shotgun_deploy/descriptor_io/appstore.py

```diff
@@ -0,0 +1,656 @@
+        self.__cache_app_store_metadata(cache_file)
+
+    def __cache_app_store_metadata(self, path):
+        """
+        Fetches metadata about the app from the toolkit app store. Writes it to disk.
+
+        :param path: Path to write the cache file to.
+        :returns: A dictionary with keys bundle and version, containing
+                  Shotgun metadata.
+        """
+        # get the appropriate shotgun app store types and fields
+        bundle_entity = self._APP_STORE_OBJECT[self._type]
+        version_entity = self._APP_STORE_VERSION[self._type]
+        link_field = self._APP_STORE_LINK[self._type]
+
+        # connect to the app store
+        (sg, _) = self.__create_sg_app_store_connection()
+
+        if self._type == self.CORE:
+            # special handling of core since it doesn't have a high-level
+            # 'bundle' entity
+            bundle = None
+            
+            version = sg.find_one(
+                constants.TANK_CORE_VERSION_ENTITY,
+                [["code", "is", self._version]],
+                ["description",
+                 "sg_detailed_release_notes",
+                 "sg_documentation",
+                 constants.TANK_CODE_PAYLOAD_FIELD]
+            )
+            if version is None:
+                raise ShotgunDeployError(
+                    "The App store does not have a version '%s' of Core!" % self._version
+                )
+            
+        else:
+            # engines, apps etc have a 'bundle level entity' in the app store,
+            # e.g. something representing the app or engine.
+            # then a version entity representing a particular version
+            bundle = sg.find_one(
+                bundle_entity,
+                [["sg_system_name", "is", self._name]],
+                ["sg_status_list", "sg_deprecation_message"]
+            )
+
+            if bundle is None:
+                raise ShotgunDeployError(
+                    "The App store does not contain an item named '%s'!" % self._name
+                )
+    
+            # now get the version
+            version = sg.find_one(
+                version_entity,
+                [[link_field, "is", bundle], ["code", "is", self._version]],
+                ["description",
+                 "sg_detailed_release_notes",
+                 "sg_documentation",
+                 constants.TANK_CODE_PAYLOAD_FIELD]
+            )
+            if version is None:
+                raise ShotgunDeployError(
+                    "The App store does not have a "
+                    "version '%s' of item '%s'!" % (self._version, self._name)
+                )
+
+        metadata = {"bundle": bundle, "version": version}
+
+        fp = open(path, "wt")
+        try:
+            pickle.dump(metadata, fp)
+            log.debug("Wrote app store cache file '%s'" % path)
+        finally:
+            fp.close()
+
+        return metadata
+
+    def _get_cache_paths(self):
```

Comment:
I was looking at that code yesterday and I was wondering if it shouldn't return only paths that exist?